buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -131,6 +133,7 @@ class ClusterFormationTasks {
     /** Adds a task to extract the elasticsearch distribution */
     static Task configureExtractTask(String name, Project project, Task setup, NodeInfo node) {
         List extractDependsOn = [project.configurations.elasticsearchDistro, setup]
+        // TODO assert that project.configurations.elasticsearchDistro.singleFile is from a project and not the cache
```

Comment:
Perhaps we can just be smarter here. If we "knew" in the method that sets up distribution that we are in elasticsearch project, we can set up a direct project dependency instead of the current dependency.